nibabel/tests/test_nifti1.py

```diff
@@ -359,7 +360,7 @@ def test_qform_sform(self):
             aff, code = out_meth(coded=True)
             assert_array_equal(aff, nice_aff)
             assert_equal(code, 2)
-            assert_array_equal(out_meth(), nice_aff) # non coded
+            assert_array_equal(out_meth(), nice_aff)  # non coded
             # Affine can also be passed if code == 0, affine will be suitably set
```

Comment:
Actually, the affine gets set to the passed affine, even if the code == 0.   The test doesn't do a good job of testing that, because the affine could have remained unchanged.  How about:

```
             another_aff = np.diag([3, 4, 5, 1])
             # Affine may be passed if code == 0, and will get set into header,
             # but the returned affine with 'coded=True' will be None.
             in_meth(another_aff, 0)
             assert_equal(out_meth(coded=True), (None, 0))  # coded -> None
             assert_array_almost_equal(out_meth(), another_aff)  # not coded -> input affine
```